vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc

```diff
@@ -99,28 +99,34 @@ LedgerImpl::~LedgerImpl() {
   }
 }
 
-void LedgerImpl::OnWalletInitializedInternal(
-    ledger::Result result,
+void LedgerImpl::OnInitialized(
+    const ledger::Result result,
     ledger::ResultCallback callback) {
   initializing_ = false;
   callback(result);
-  if (result == ledger::Result::LEDGER_OK ||
-      result == ledger::Result::WALLET_CREATED) {
+  if (result == ledger::Result::LEDGER_OK) {
     initialized_ = true;
-    bat_publisher_->SetPublisherServerListTimer(GetRewardsMainEnabled());
-    bat_contribution_->SetReconcileTimer();
-    bat_promotion_->Refresh(false);
-    bat_contribution_->Initialize();
-    bat_promotion_->Initialize();
-    bat_api_->Initialize();
-    braveledger_recovery::Check(this);
-
-    SetConfirmationsWalletInfo();
+    StartServices();
   } else {
     BLOG(0, "Failed to initialize wallet " << result);
   }
 }
 
+void LedgerImpl::StartServices() {
+  if (!IsWalletCreated()) {
+    return;
+  }
+
+  bat_publisher_->SetPublisherServerListTimer(GetRewardsMainEnabled());
+  bat_contribution_->SetReconcileTimer();
+  bat_promotion_->Refresh(false);
+  bat_contribution_->Initialize();
+  bat_promotion_->Initialize();
+  bat_api_->Initialize();
+  SetConfirmationsWalletInfo();
+  braveledger_recovery::Check(this);
```

Comment:
This line was previously before `SetConfirmationsWalletInfo`. Was the reordering intentional?

Reply:
not really, but it makes no difference